### PR TITLE
Support for Unity quota management

### PR DESCRIPTION
**Pull request by rajendraindukuri**
This PR covers support for Unity Quota management:
1. CRUD operations (whichever supported by API) for userQuota, treeQuota ,quotaConfig components
2. Unit test coverage for the same

### CODE REVIEW

File: storops_test/unity/resource/test_system.py [hunk 1]
```diff
@@ -329,6 +336,270 @@ def test_create_nas_server(self):
...
+        assert_that(ret, instance_of(UnityUserQuota))
+        assert_that(ret.id, equal_to('userquota_171798692187_3_3'))
+        assert_that(ret.existed, equal_to(True))
+
+    @patch_rest
+    def test_create_user_quota_on_tree_quota_fs_invalid(self):
+        def f():
+            unity = t_unity()
+            unity.create_user_quota(
+                                  filesystem_id='fs_99',
```
Comment (yong-huang) [inline storops_test/unity/resource/test_system.py:399]: Indentation

File: storops_test/unity/resource/test_system.py [hunk 2]
```diff
@@ -329,6 +336,270 @@ def test_create_nas_server(self):
...
+                filesystem_id='fs_99', hard_limit=9663676416,
+                soft_limit=3221225472, uid=3)
+
+        assert_that(f, raises(UnityResourceNotFoundError))
+
+    @patch_rest
+    def test_create_user_quota_on_tree_quota(self):
+        unity = t_unity()
+        ret = unity.create_user_quota(
+                                    filesystem_id='fs_2',
```
Comment (yong-huang) [inline storops_test/unity/resource/test_system.py:385]: Indentation

File: storops_test/unity/resource/test_system.py [hunk 3]
```diff
@@ -329,6 +336,270 @@ def test_create_nas_server(self):
...
+                                  uid=3)
+
+        assert_that(f, raises(UnityResourceNotFoundError))
+
+    @patch_rest
+    def test_create_user_quota_on_tree_quota_tq_invalid(self):
+        def f():
+            unity = t_unity()
+            unity.create_user_quota(
+                                  filesystem_id='fs_2',
```
Comment (yong-huang) [inline storops_test/unity/resource/test_system.py:412]: Indentation

File: storops_test/unity/resource/test_system.py [hunk 4]
```diff
@@ -329,6 +336,270 @@ def test_create_nas_server(self):
...
+                                  uid=3)
+
+        assert_that(f, raises(UnityResourceNotFoundError))
+
+    @patch_rest
+    def test_create_user_quota_on_tree_quota_tq_n_fs_invalid(self):
+        def f():
+            unity = t_unity()
+            unity.create_user_quota(
+                                  filesystem_id='fs_99',
```
Comment (yong-huang) [inline storops_test/unity/resource/test_system.py:425]: Indentation

File: storops_test/unity/resource/test_system.py [hunk 5]
```diff
@@ -329,6 +336,270 @@ def test_create_nas_server(self):
...
+                default_hard_limit=8589934592,
+                default_soft_limit=2147483648)
+        assert_that(resp.is_ok(), equal_to(True))
+
+    @patch_rest
+    def test_modify_quota_config_negative(self):
+        def f():
+            unity = t_unity()
+            unity.modify_quota_config(
+                                      quota_config_id='abc',
```
Comment (yong-huang) [inline storops_test/unity/resource/test_system.py:576]: Indentation

**Review by yong-huang — CHANGES_REQUESTED** [review]

File: storops_test/unity/rest_data/quotaConfig/quota_config_not_found.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "content": {
+    "existed": false,
```
[the remark this answers is not in the document]
Comment (rajendraindukuri) [reply storops_test/unity/rest_data/quotaConfig/quota_config_not_found.json:3]: This is a storops response.

File: storops_test/unity/rest_data/treeQuota/tree_quota_not_found.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+  "content": {
+    "existed": false,
```
[the remark this answers is not in the document]
Comment (rajendraindukuri) [reply storops_test/unity/rest_data/treeQuota/tree_quota_not_found.json:3]: This is a storops response

File: storops_test/unity/rest_data/userQuota/user_quota_not_found.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{
+  "content": {
+    "existed": false,
```
[the remark this answers is not in the document]
Comment (rajendraindukuri) [reply storops_test/unity/rest_data/userQuota/user_quota_not_found.json:3]: This is a storops response

File: storops_test/unity/resource/test_user_quota.py [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+    @patch_rest
+    def test_create_user_quota_negative(self):
+        def f():
+            UnityUserQuota.create(cli=t_rest(),
+                                  filesystem_id='fs_99',
+                                  hard_limit=9663676416,
+                                  soft_limit=3221225472,
+                                  uid=3)
+
+        assert_that(f, raises(UnityResourceNotFoundError))
```
[the remark this answers is not in the document]
Comment (rajendraindukuri) [reply storops_test/unity/resource/test_user_quota.py:73]: Ya @yong-huang . Just reused the existing filesystem rest data

File: storops/unity/resource/tree_quota.py [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+            tree_quota specified
+        :return: None
+        """
+        tree_quota = UnityTreeQuota.get(_id=tree_quota_id, cli=cli)
+        if not tree_quota.existed:
+            raise UnityResourceNotFoundError(
+                'cannot find tree_quota {}.'.format(tree_quota_id))
+
+        req_body = cli.make_body(
+            allow_empty=False,
```
[the remark this answers is not in the document]
Comment (rajendraindukuri) [reply storops/unity/resource/tree_quota.py]: Hi @yong-huang , For now removed this attribute as by default it is false and tested all the scenarios.. it looks fine. But I did not fully understand the significance of this attribute, if you can help me with understanding the same I will try to use it accordingly. Thanks.

Same for all the similar comments

[the remark this answers is not in the document]
Comment (yong-huang) [reply storops/unity/resource/tree_quota.py]: It ok if all tests passed. It only impacts the empty list/dict in the parameters.

**Review by yong-huang — APPROVED** [review]

**Pull request merged**